slatedb/src/retention_iterator.rs

```diff
@@ -72,20 +77,30 @@ impl<T: KeyValueIterator> RetentionIterator<T> {
         retention_timeout: Option<Duration>,
         retention_min_seq: Option<u64>,
         filter_tombstone: bool,
+        sequence_tracker: Arc<SequenceTracker>,
     ) -> BTreeMap<Reverse<u64>, RowEntry> {
         let mut filtered_versions = BTreeMap::new();
+        let now = system_clock.now().timestamp_millis();
         for (idx, (_, entry)) in versions.into_iter().enumerate() {
             // always keep the latest version (idx == 0), for older versions, check if they are within retention window.
             // retention window is defined by either timeout or seq, or both. if both are not set, only the latest version
             // is kept.
             let in_retention_window_by_time = retention_timeout
                 .map(|timeout| {
-                    let create_ts = entry
-                        .create_ts
-                        .expect("a record with no create_ts should not happen");
-                    // TODO: This is wrong! We're mixing logical (create_ts) and physical (system_clock) timestamps.
+                    let create_sys_ts = sequence_tracker
+                        // we use RoundUp so that we are less aggressive about filtering
+                        // rows (e.g. if the requested window is 10min and the closes recorded
+                        // sequence numbers are 9m30s and 10m30s ago for upper/lower rounding
+                        // respectively we will conservatively select to keep this record in
+                        // the filtered results)
+                        .find_ts(entry.seq, FindOption::RoundUp)
```

Comment:
Want to make sure I understand this correctly.

> if the requested window is 10min...

By "requested window", you mean if the actual timestamp for the `entry.seq` is `10m`, but that isn't stored because we're best effort on seqnum<>timestamp mapping, right?

> the closes recorded sequence numbers are 9m30s and 10m30s ago for upper/lower rounding respectively...

Wouldn't this be lower/upper rounding respectively? I'm looking at:

```
pub(crate) enum FindOption {
    /// Round up to the next higher value when no exact match is found.
    RoundUp,
    /// Round down to the next lower value when no exact match is found.
    RoundDown,
}
```

If my seqnum is 123, and I have 120=9m30s and 125=10m30s, RoundUp should return 10m30s, right?